### DISCUSSION TITLE
Helm chart and GitRepository Intervals: How do they work?

### DISCUSSION THREAD

**Discussion by PaulSxxxs** [fluxcd/flux2#4247] [Q&A] [answered]
I have upwards of 200 helm charts and currently have some issues with flux, it seems to be failing at some point dealing with git repository and image automation failing with "object not found:".

I'm investigating if we're possibly flooding bitbucket with too many requests and wondered how flux handles this config:
e.g. will flux iterate through every helm chart, every 3 minutes and open/close a connection to the git repo each time? or is flux intelligent enough to handle all iterations through the charts in one connection to bitbucket ... and then proceed to do some more git actions requiring communicating with bitbucket for image automation etc?

![image](https://github.com/fluxcd/flux2/assets/145309332/208d51bc-55cb-4f87-95b3-8dfe4ff49238)

**Comment by stefanprodan** [comment, marked as answer]
For each chart, Flux does the equivalent of git-ls-remote, and it only pulls the Git repo contents when a new commit is discovered. Listing the commits is a lightweight operation, but it does open a new TCP connection every time. I suggest you set a higher interval in the HelmReleases, Flux is event-based, it will detect changes upstream using the GitRepository interval and will automatically trigger all releases on its own when the new commits are discovered.
